org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/DirectedGraphLayoutAlgorithm.java

```diff
@@ -34,20 +34,14 @@ public void visit(DirectedGraph graph) {
 				field = DirectedGraphLayout.class.getDeclaredField("steps");
```

Comment:
should we add a protected getter to the steps so that we don't need reflection here?

Reply:
I'm not sure if that would really work... "steps" stores objects of type GraphVisitor, which is a package-private class in draw2d. With a getter, this class would now be exposed.
As an example, the method call in DirectedGraphLayoutAlgorithm would still have to look like `Deque<?> parts = getParts()`, simply because GraphVisitor is not visible.

Reply:
I meant to add a protected getSteps method to DirectedGraph Layout so that Children like this ExtenddedDirectGraphLayout could access the steps. But I'm not sure if we want to expose it like that. As you say this has implications.

Reply:
Let me rephrase it: What should getSteps() return? It can't be Deque<GraphVisitor> because the GraphVisitor is package-private class. So the only other alternatives are Deque<?> or Deque<Object>, both of which I'm not really happy with...

Reply:
Ah now I get it. Yes a getter makes here then no sense at all.